corehq/apps/sso/models.py

```diff
@@ -143,6 +143,13 @@ class IdentityProvider(models.Model):
     last_modified = models.DateTimeField(auto_now=True)
     last_modified_by = models.EmailField()
 
+    # for auto-deactivation web user purposes
+    enable_user_deactivation = models.BooleanField(default=False)
+    api_host = models.TextField(default="")  # tenant id
+    api_id = models.TextField(default="")  # application (client) id in Azure AD
+    api_secret = models.TextField(default="")
+    date_api_secret_expiration = models.DateField(blank=True, null=True)
```

Comment:
Does allowing `blank` and `null` support a specific use case? I'm looking at [the django docs](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.Field.null), which mention:
> In most cases, it’s redundant to have two possible values for “no data;”

I realize the docs also say
> you will also need to set blank=True if you wish to permit empty values in forms, as the null parameter only affects database storage

but I was curious if we could avoid the concept of empty values here when `enable_user_deactivation` is `False`

Reply:
When `enable_user_deactivation` is  `False`, I interpret that to mean we don't supply _any_ value to the fields controlled by it. Assigning an empty value is a bit different than not assigning a value at all